src/AccelStepperFirmata.cpp

```diff
@@ -0,0 +1,429 @@
+        if (interface & 0x01) {
+          enablePin = argv[index++];
+          if (Firmata.getPinMode(enablePin) == PIN_MODE_IGNORE)
+            return false;
+        }
+
+        // Instantiate our stepper
+        if (wireCount == 1) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::DRIVER, stepOrMotorPin1, directionOrMotorPin2);
+        } else if (wireCount == 2) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::FULL2WIRE, stepOrMotorPin1, directionOrMotorPin2);
+        } else if (wireCount == 3 && stepType == STEP_TYPE_WHOLE) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::FULL3WIRE, stepOrMotorPin1, directionOrMotorPin2, motorPin3);
+        } else if (wireCount == 3 && stepType == STEP_TYPE_HALF) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::HALF3WIRE, stepOrMotorPin1, directionOrMotorPin2, motorPin3);
+        } else if (wireCount == 4 && stepType == STEP_TYPE_WHOLE) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::FULL4WIRE, stepOrMotorPin1, directionOrMotorPin2, motorPin3, motorPin4, false);
+        } else if (wireCount == 4 && stepType == STEP_TYPE_HALF) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::HALF4WIRE, stepOrMotorPin1, directionOrMotorPin2, motorPin3, motorPin4, false);
+        }
+        
+        // If there is still another byte to read we must be inverting some pins
+        if (argc >= index) {
+          invertPins = argv[index];
+          if (wireCount == 1) {
+             stepper[deviceNum]->setPinsInverted(invertPins & 0x01, invertPins >> 1 & 0x01, invertPins >> 4 & 0x01);
+          } else {
+            stepper[deviceNum]->setPinsInverted(invertPins & 0x01, invertPins >> 1 & 0x01, invertPins >> 2 & 0x01, invertPins >> 3 & 0x01, invertPins >> 4 & 0x01);
+          }
+        }
+
+        if (interface & 0x01) {
+          stepper[deviceNum]->setEnablePin(enablePin);
+        }
+
+        /*
+          Default to no acceleration. We set the acceleration value high enough that our speed is 
+          reached on the first step of a movement. 
+          More info about this hack in ACCELSTEPPER_SET_ACCELERATION.
+
+          The lines where we are setting the speed twice are necessary because if the max speed doesn't change 
+          from the default value then our time to next step does not get computed after raising the acceleration.
+        */
+        stepper[deviceNum]->setMaxSpeed(2.0);
+        stepper[deviceNum]->setMaxSpeed(1.0);
+        stepper[deviceNum]->setAcceleration(MAX_ACCELERATION);
+
+        isRunning[deviceNum] = false;
+
+      }
+
+      else if (stepCommand == ACCELSTEPPER_STEP) {
+
+        numSteps = decode32BitSignedInteger(argv[2], argv[3], argv[4], argv[5], argv[6]);
+
+        if (stepper[deviceNum]) {
+          stepper[deviceNum]->move(numSteps);
+          isRunning[deviceNum] = true;
+        }
+
+      }
+
+      else if (stepCommand == ACCELSTEPPER_ZERO) {
```

Comment:
What is the use case of Zero? Currently it will stop the motor when called since a position of zero will cause the AccelStepper run method to return false. The only difference between Zero and Stop, is that when calling Zero, the position reported is 0 vs Stop where the position is the current step count. I'm assuming the use case for Zero however is to typically call it while the motor is not running. Is this a correct assumption?

Reply:
This is usually called after an initial positioning routine has run at start up. 

accelStepper instances have an absolute position counter. When an accelStepper is initialized the counter starts at zero. Calling stepper zero resets that value. 

setCurrentPosition explicitly sets the speed of the motor to zero. I assume this is because any already running movements could behave unpredictably.